vtgate_gateway_collectd.py

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/python
+
+import time
+import util
+import mock
+
+NAME = 'vtgate'
+
+#Module that generates metric using the vtgate /debug/gateways endpoint
+#Metrics generated
+# For all keyspace shards  servingMaster: 1/0 indicating whether there is a Master tablet up and serving
+class VtgateGateway(util.BaseCollector):
+    def __init__(self, collectd, json_provider=None, verbose=False, interval=None):
+        super(VtgateGateway, self).__init__(collectd, NAME, 15001, json_provider, verbose, interval)
+
+    def configure_callback(self, conf):
+        super(VtgateGateway, self).configure_callback(conf)
+        self.register_read_callback()
+
+    def process_data(self, json_data):
+       for keyspaceShardTabletType, tablets in json_data.items():
+          if "MASTER" in keyspaceShardTabletType:
```

Comment:
so if we don't find a primary, we send no signal, and thus 0 or null mean no serving primary?

Reply:
to some extend yes. If there is a Pimary tablet in the topology we'll always find something in the gateway response and send 1/0.
If for some reason we remove primary pod, we won't send any metrics so we have to rely on monitoring to catch this.